### ISSUE TITLE
The recommendation loss calculation seems different from the original paper?

### ISSUE THREAD

**Issue by ptkenny** [ghdtjr/A-LLMRec#4]
Hey, I'm currently trying to go through this implementation and understand how everything works. Looking at `A_llmrec_model` here: https://github.com/ghdtjr/A-LLMRec/blob/master/models/a_llmrec_model.py#L59

If I'm understanding the paper correctly, it seems like the BCE loss should be configured with `reduction='sum'`? From equation 5:

<img width="512" alt="Screenshot 2024-08-07 at 2 58 42 PM" src="https://github.com/user-attachments/assets/adf15f21-7111-4adf-bf03-b88dd4e29db3">

From the pytorch docs, the default is reduction is `mean` which doesn't seem to be the intended computation here: https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html

Please let me know if I'm misunderstanding something or if this is otherwise intentional. Thanks!

**Comment by ptkenny** [comment]
Also, is the dot product missing here or is it done somewhere and I'm missing it?

**Comment by ghdtjr** [comment]
We followed the implementation details with the SASRec. It seems that the difference between the 'sum' and 'mean' is only about the scale. You can implement one of them. 

If you are interested in the detailed implementation please reference the below links.

https://github.com/pmixer/SASRec.pytorch
https://arxiv.org/abs/1808.09781
